lib/safe-pg-migrations/plugins/statement_insurer.rb

```diff
@@ -10,11 +10,13 @@ module StatementInsurer
       end
     end
 
-    def add_column(table_name, column_name, type, **options) # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    def add_column(table_name, column_name, type, **options)
       need_default_value_backfill = SafePgMigrations.pg_version_num < PG_11_VERSION_NUM
+      apply_not_null_constraint_separately = SafePgMigrations.pg_version_num < PG_11_VERSION_NUM
 
       default = options.delete(:default) if need_default_value_backfill
-      null = options.delete(:null)
+      null = options.delete(:null) if apply_not_null_constraint_separately
```

Comment:
Actually, now that `null` is natively handled by ActiveRecord's `add_column`, we  do nothing if PG version is recent enough. We could simplify a lot this method:

```rb

def add_column(table_name, column_name, type, **options)
    return super if SafePgMigrations.pg_version_num >= PG_11_VERSION_NUM
    
    default = options.delete(:default)
    null = options.delete(:null)
    # ...
end
```

Would you like to handle this refactoring as well?

Reply:
That totally makes sense! I can make this change shortly.

Reply:
Done! I also added a few more tests for the older version of Postgres to cover a few more variations of arguments for the older versions since I was changing that logic a little more.

Let me know if we need to do anything else here!